operator/internal/handlers/lokistack_create_or_update_test.go

```diff
@@ -655,14 +674,19 @@ func TestCreateOrUpdateLokiStack_WhenMissingSecret_SetDegraded(t *testing.T) {
 
 	k.StatusStub = func() client.StatusWriter { return sw }
 
-	err := handlers.CreateOrUpdateLokiStack(context.TODO(), r, k, scheme, flags)
+	degraded, err := handlers.CreateOrUpdateLokiStack(context.TODO(), r, k, scheme, flags)
 
-	// make sure error is returned to re-trigger reconciliation
+	// make sure error is not returned to re-trigger reconciliation
 	require.NoError(t, err)
+	// make sure degraded is non-nil
+	require.NotNil(t, degraded)
 
-	// make sure status and status-update calls
-	require.NotZero(t, k.StatusCallCount())
-	require.NotZero(t, sw.UpdateCallCount())
+	require.Equal(t, msg, degraded.Message)
+	require.Equal(t, reason, degraded.Reason)
```

Comment:
Does it make sense to define a `DegradedCondition` instead and compare the whole struct in one assertion?